n3fit/src/n3fit/model_trainer.py

```diff
@@ -853,7 +862,16 @@ def hyperparametrizable(self, params):
         exp_models = []
 
         # Generate the grid in x, note this is the same for all partitions
-        xinput = self._xgrid_generation()
+        xinput, inputs_unique = self._xgrid_generation()
+        
+        # compute photon:
+        photon=Photon(theoryid=self.theoryid, fiatlux_runcard=self.fiatlux_runcard)
+        photon_array = np.array([])
+        for i in inputs_unique:
+            xgrid = i[0]
+            photon_array = np.append(photon_array, photon.photon_fitting_scale(xgrid))
```

Comment:
This can be done a lot cleaner, e.g.:
`photon_array = np.array([photon.photon_fitting_scale(xgrid[0]) for xgrid in inputs_unique])`